### PR TITLE
using co.js rather than yield util

### PR DESCRIPTION
**Pull request by damnhipster**

### CODE REVIEW

File: scripts/build [hunk 1]
```diff
@@ -3,6 +3,7 @@
 require('../src/global');
 
 var cli = require('commander');
+var yield = require('co');
```
Comment (bneiluj) [inline scripts/build]: I don't think it's a good practice to declare `yield` from `co` and then use it as it was a `yield` feature!

File: scripts/build [hunk 2]
```diff
@@ -21,10 +22,13 @@ if (typeof styles === 'undefined') {
...
 }
 
-utils.yield(function * main() {
+yield(function * main() {
   var list = yield styles.map(function(style) {
     return { name: style }
   });
   yield * build(list);
   console.log('Generated', styles);
+}).catch(function(err) {
```
Comment (bneiluj) [inline scripts/build]: you should be yielding inside the generator function - wrapped with `co` ... it makes the flow easier to understand.

File: src/core/utils.js [hunk 3]
```diff
@@ -57,33 +57,11 @@ function writeFile(path, data) {
 	});
 }
 
-function yield(gen) {
```
Comment (bneiluj) [inline src/core/utils.js]: 👍

Comment (bneiluj) [inline src/core/utils.js]: finally got rid off this fancy function 🎱

File: scripts/build (hunk 2 above)
Comment (damnhipster) [inline scripts/build]: Why surround it with another generator function? `build` is already a generator function, not a `Promise`.

The only reason to surround it with another is if I wanted to add exception handling, but that's done via the catch statement.

File: scripts/build (hunk 1 above)
Comment (damnhipster) [inline scripts/build]: I had to dig into the koa.js code to find it about CO in the first place, and it didn't make it very easy to find. One of the reasons was that it was referenced as CO, which is not easy to understand at all.

**Pull request merged**